MoreLinq.Test/EndsWithTest.cs

```diff
@@ -0,0 +1,79 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2008 Jonathan Skeet. All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    public class EndsWithTest
+    {
+        [TestCase(null, null)]
+        [TestCase(null, new[] {1})]
+        [TestCase(new[] {1}, null)]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void EndsWithThrowsIfFirstOrSecondAreNull(IEnumerable<int> first, IEnumerable<int> second)
+        {
+            first.EndsWith(second);
+        }
+
+        [TestCase(new[] {1, 2, 3}, new[] {2, 3}, Result = true)]
+        [TestCase(new[] {1, 2, 3}, new[] {1, 2, 3}, Result = true)]
+        [TestCase(new[] {1, 2, 3}, new[] {0, 1, 2, 3}, Result = false)]
+        public bool EndsWithWithIntegers(IEnumerable<int> first, IEnumerable<int> second)
+        {
+            return first.EndsWith(second);
+        }
+
+        [TestCase(new[] {'1', '2', '3'}, new[] {'2', '3'}, Result = true)]
+        [TestCase(new[] {'1', '2', '3'}, new[] {'1', '2', '3'}, Result = true)]
+        [TestCase(new[] {'1', '2', '3'}, new[] {'0', '1', '2', '3'}, Result = false)]
+        public bool EndsWithWithChars(IEnumerable<char> first, IEnumerable<char> second)
+        {
+            return first.EndsWith(second);
+        }
+
+        [TestCase("123", "23", Result = true)]
+        [TestCase("123", "123", Result = true)]
+        [TestCase("123", "0123", Result = false)]
+        public bool EndsWithWithStrings(string first, string second)
+        {
+            // Avoid ambiguity with String.EndsWith()
+            return MoreEnumerable.EndsWith(first, second);
+        }
+
+        [Test]
+        public void EndsWithReturnsTrueIfBothEmpty()
+        {
+            new int[0].EndsWith(new int[0]);
+        }
+
+        [Test]
+        public void EndsWithReturnsFalseIfOnlyFirstIsEmpty()
+        {
+            new int[0].EndsWith(new[] {1,2,3});
+        }
+
+        [Test]
+        public void EndsWithReturnsFalseIfOnlySecondIsEmpty()
+        {
+            new[] {1,2,3}.EndsWith(new int[0]);
```

Comment:
I just realized the implementation was already like you wanted, but the tests were lacking the asserts :-)